lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +198,89 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        article = article_soup.find_all('div', {'itemprop': 'articleBody'})[0]
+        article_list = article.find_all('p')
+        paragraphs = [par.text for par in article_list]
+        self.article.text = '\n'.join(paragraphs)
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        article = article_soup.find_all('div', {'itemprop': 'headline'})[0]
+        article_title = article.find_all('h1')[0]
+        self.article.title = article_title.text
+        article_authors = article_soup.find_all('span', {'itemprop': 'author'})[0]
+        authors = article_authors.find_all('span', {'itemprop': 'name'})
+        if authors:
+            self.article.author = [author.text for author in authors]
+        else:
+            self.article.author = ['NOT FOUND']
+        try:
+            article_tags = article_soup.find_all('ul', {'itemprop': 'keywords'})[0]
+            article_tags_li = article_tags.find_all('li')
+            self.article.topics = [tag.find_all('a')[0].text for tag in article_tags_li]
+        except IndexError:
+            self.article.topics = []
+        article_date = article_soup.find_all('span', {'class': 'news-datetime mb-10 mr-20'})[0]
```

Comment:
is there a better way to find date? sequence of classes is fragile.

Reply:
Done - I used itemprop